### PR TITLE
feat(odsp-driver): Add logging property if doc is in dogfood or MSIT server environments

### PR DESCRIPTION
**Pull request by markfields**
## Description

Fixes [AB#16011](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/16011)

Sometimes we are investigating a live site incident with our 1P customers, and it would be helpful to know which service environment (SharePoint "Farm") the document belongs to, since different settings or code can be deployed to these independently.  We can get a reasonable proxy of this by inspecting the domain of the URL against some known patterns.

We are computing/logging this for trees-latest and create-New events - should typically be once per session.

### CODE REVIEW

File: packages/drivers/odsp-driver/src/createFile/createFile.ts [hunk 1]
```diff
@@ -193,10 +193,16 @@ export async function createNewEmptyFluidFile(
...
 			"CreateNewFile",
 		);
+		const internalFarmType = checkForInternalFarmType(newFileInfo.siteUrl);
 
 		return PerformanceEvent.timedExecAsync(
 			logger,
-			{ eventName: "createNewEmptyFile" },
+			{
+				eventName: "createNewEmptyFile",
+				details: {
```
Comment (jatgarg) [inline packages/drivers/odsp-driver/src/createFile/createFile.ts:202]: Don't we stringify the details prop bag?

Comment (jatgarg) [reply packages/drivers/odsp-driver/src/createFile/createFile.ts:202]: Also, createFile has 2 functions. This function handles where we create the empty file and there is one where we create file with intial summary.

Comment (markfields) [reply packages/drivers/odsp-driver/src/createFile/createFile.ts:202]: > Don't we stringify the details prop bag?

It happens automatically now, no need to call stringify here.

File: packages/drivers/odsp-driver/src/createFile/createFile.ts [hunk 2]
```diff
@@ -26,7 +26,7 @@ import { createOdspUrl } from "./../createOdspUrl.js";
 import { EpochTracker } from "./../epochTracker.js";
 import { getHeadersWithAuth } from "./../getUrlAndHeadersWithAuth.js";
 import { OdspDriverUrlResolver } from "./../odspDriverUrlResolver.js";
-import { getApiRoot } from "./../odspUrlHelper.js";
+import { checkForInternalFarmType, getApiRoot } from "./../odspUrlHelper.js";
```
Comment (jatgarg) [inline packages/drivers/odsp-driver/src/createFile/createFile.ts:29]: SHould we rename this to checkForServiceFarmType or checkForServerFarmType? Internal does not make sense to me.

**Review by jatgarg — APPROVED** [review]

**Pull request merged**